packages/editor/src/components/block-settings-menu/with-block-settings-actions.js

```diff
@@ -1,56 +1,55 @@
 /**
  * External dependencies
  */
-import { flow, noop, last, every, first, castArray } from 'lodash';
+import { castArray, first, last, every } from 'lodash';
 
 /**
  * WordPress dependencies
  */
-import { __ } from '@wordpress/i18n';
-import { IconButton } from '@wordpress/components';
 import { compose } from '@wordpress/compose';
 import { withSelect, withDispatch } from '@wordpress/data';
 import { cloneBlock, hasBlockSupport } from '@wordpress/blocks';
+import { rawShortcut, displayShortcut } from '@wordpress/keycodes';
 
-export function BlockDuplicateButton( { blocks, onDuplicate, onClick = noop, isLocked, small = false, role } ) {
-	const canDuplicate = every( blocks, ( block ) => {
-		return hasBlockSupport( block.name, 'multiple', true );
-	} );
-	if ( isLocked || ! canDuplicate ) {
-		return null;
-	}
+const shortcuts = {
+	duplicate: {
+		raw: rawShortcut.primaryShift( 'd' ),
+		display: displayShortcut.primaryShift( 'd' ),
+	},
+	remove: {
+		raw: rawShortcut.primary( 'del' ),
+		display: displayShortcut.primary( 'del' ),
+	},
+};
 
-	const label = __( 'Duplicate' );
-
-	return (
-		<IconButton
-			className="editor-block-settings-menu__control"
-			onClick={ flow( onDuplicate, onClick ) }
-			icon="admin-page"
-			label={ small ? label : undefined }
-			role={ role }
-		>
-			{ ! small && label }
-		</IconButton>
-	);
-}
-
-export default compose(
+const withBlockSettingsActions = compose( [
 	withSelect( ( select, { clientIds, rootClientId } ) => {
 		const {
 			getBlocksByClientId,
 			getBlockIndex,
 			getTemplateLock,
 		} = select( 'core/editor' );
 
+		const blocks = getBlocksByClientId( clientIds );
+
+		const canDuplicate = every( blocks, ( block ) => {
+			return !! block && hasBlockSupport( block.name, 'multiple', true );
+		} );
+
 		return {
-			blocks: getBlocksByClientId( clientIds ),
 			index: getBlockIndex( last( castArray( clientIds ) ), rootClientId ),
 			isLocked: !! getTemplateLock( rootClientId ),
+			blocks,
+			canDuplicate,
+			shortcuts,
```

Comment:
Why have `withSelect` pass `shortcuts` through as a prop here? Would it make sense for `shortcuts` to just be a regular top-level variable, since it never changes when the store updates?

Reply:
Now used as just a regular variable since 8db7543